tests/net/lib/coap_client/src/main.c

```diff
@@ -357,6 +387,38 @@ ZTEST(coap_client, test_get_request)
 	zassert_equal(last_response_code, COAP_RESPONSE_CODE_OK, "Unexpected response");
 }
 
+ZTEST(coap_client, test_resend_request)
+{
+	int ret = 0;
+	struct sockaddr address = {0};
+	struct coap_client_request client_request = {
+		.method = COAP_METHOD_GET,
+		.confirmable = true,
+		.path = test_path,
+		.fmt = COAP_CONTENT_FORMAT_TEXT_PLAIN,
+		.cb = coap_callback,
+		.payload = NULL,
+		.len = 0
+	};
+
+	client_request.payload = short_payload;
+	client_request.len = strlen(short_payload);
+
+	z_impl_zsock_recvfrom_fake.custom_fake = z_impl_zsock_recvfrom_custom_fake_delayed_response;
+
+	k_sleep(K_MSEC(1));
+
+	LOG_INF("Send request");
+	ret = coap_client_req(&client, 0, &address, &client_request, -1);
+	zassert_true(ret >= 0, "Sending request failed, %d", ret);
+	k_sleep(K_MSEC(3000));
```

Comment:
Do we really need such a long delays? There is now 5 seconds added by this commit, that feels excessive.
If we do not know exactly how long to wait, we could perhaps use k_sem to wait something to happen here and then release it in other part of the test when we know we can proceed.

Reply:
Maybe we could just lower `CONFIG_COAP_INIT_ACK_TIMEOUT_MS` to test in tighter timeframes?

Reply:
Good point. I'll lower CONFIG_COAP_INIT_ACK_TIMEOUT_MS and delays.